blockchain/blockchain.go

```diff
@@ -240,9 +254,228 @@ func NewBlockChain(db database.DBManager, cacheConfig *CacheConfig, chainConfig
 	// Take ownership of this particular state
 	go bc.update()
 	go bc.gcCachedNodeLoop()
+	go bc.migrationLoop()
 	return bc, nil
 }
 
+type stateTrieMigrationDB struct {
+	database.DBManager
+}
+
+func (td *stateTrieMigrationDB) ReadCachedTrieNode(hash common.Hash) ([]byte, error) {
+	return td.ReadCachedTrieNodeFromNew(hash)
+}
+func (td *stateTrieMigrationDB) ReadCachedTrieNodePreimage(secureKey []byte) ([]byte, error) {
+	return td.ReadCachedTrieNodePreimageFromNew(secureKey)
+}
+
+func (td *stateTrieMigrationDB) ReadStateTrieNode(key []byte) ([]byte, error) {
+	return td.ReadStateTrieNodeFromNew(key)
+}
+
+func (td *stateTrieMigrationDB) HasStateTrieNode(key []byte) (bool, error) {
+	return td.HasStateTrieNodeFromNew(key)
+}
+
+func (td *stateTrieMigrationDB) ReadPreimage(hash common.Hash) []byte {
+	return td.ReadPreimageFromNew(hash)
+}
+
+func (bc *BlockChain) migrateState(rootHash common.Hash) error {
```

Comment:
How about `startStateMigration`?

Reply:
There is `StartStateMigration` function to be called by API.
So I defined the function like that.

Reply:
Actually, that is the reason why I suggested the name, `startStateMigration`.
`StartStateMigration` is public function that is called as an API. `StartStateMigration` indirectly calls `migrateState` and `migrateState` is private function that is actually doing the migration.
It is better to say "`StartStateMigration` calls `startStateMigration`" than 
"`StartStateMigration` calls `migrateState`". What do you think?

Reply:
Well, I think "StartStateMigration calls migrateState" is better. 
"StartStateMigration calls startStateMigration" looks weird to me.

I intended like below.
`StartStateMigration` starts the migration.
`migrationState` : works some job for migration.